manticore/utils/log.py

```diff
@@ -11,70 +11,95 @@ def filter(self, record):
             record.stateid = '[%d]' % self.stateid
         else:
             record.stateid = ''
+
+        # Summarize logger name
+        components = record.name.split('.')
+        name = '.'.join(c[0] for c in components[:-1]) + '.' + components[-1]
+        record.name = name
+
         return True
 
-loggers = ['MANTICORE',
-           'VISITOR',
-           'EXECUTOR',
-           'CPU',
-           'REGISTERS',
-           'SMT',
-           'MEMORY',
-           'PLATFORM']
 manticore_verbosity = 0
+all_loggers = []
 
 def init_logging():
+    global all_loggers
+    all_loggers = logging.getLogger().manager.loggerDict.keys()
+        
     ctxfilter = ContextFilter()
     logfmt = ("%(asctime)s: [%(process)d]%(stateid)s %(name)s:%(levelname)s:"
               " %(message)s")
     logging.basicConfig(format=logfmt, stream=sys.stdout, level=logging.ERROR)
-    for l in loggers:
-        logging.getLogger(l).setLevel(logging.WARNING)
-        logging.getLogger(l).addFilter(ctxfilter)
-        logging.getLogger(l).setState = types.MethodType(loggerSetState,
-                                                         logging.getLogger(l))
+    for name in logging.getLogger().manager.loggerDict.keys():
+        logger = logging.getLogger(name)
+        if not name.startswith('manticore'):
+            next
+        logger.setLevel(logging.WARNING)
+        logger.addFilter(ctxfilter)
+        logger.setState = types.MethodType(loggerSetState, logger)
 
 def loggerSetState(logger, stateid):
     logger.filters[0].stateid = stateid
 
 def set_verbosity(setting):
-    global manticore_verbosity
-    zero = map(lambda x: (x, logging.WARNING), loggers)
+    global manticore_verbosity, all_loggers
+    zero = map(lambda x: (x, logging.WARNING), all_loggers)
     levels = [
         # 0
         zero,
         # 1
         [
-            ('MANTICORE', logging.INFO)
+            ('manticore.manticore', logging.INFO)
         ],
         # 2 (-v)
         [
-            ('EXECUTOR', logging.INFO),
-            ('PLATFORM', logging.DEBUG)
+            ('manticore.core.executor', logging.INFO),
+            ('manticore.platforms.*', logging.DEBUG)
         ],
         # 3 (-vv)
         [
-            ('CPU', logging.DEBUG)
+            ('manticore.core.cpu.*', logging.DEBUG)
         ],
         # 4 (-vvv)
         [
-            ('MEMORY', logging.DEBUG),
-            ('CPU', logging.DEBUG),
-            ('REGISTERS', logging.DEBUG)
+            ('manticore.core.memory', logging.DEBUG),
+            ('manticore.core.cpu.*', logging.DEBUG),
+            ('manticore.core.cpu.*.registers', logging.DEBUG)
         ],
         # 5 (-vvvv)
         [
-            ('MANTICORE', logging.DEBUG),
-            ('SMT', logging.DEBUG)
+            ('manticore.manticore', logging.DEBUG),
+            ('manticore.core.smtlib', logging.DEBUG),
+            ('manticore.core.smtlib.*', logging.DEBUG)
          ]
     ]
 
+    def match(name, pattern):
```

Comment:
Do we need this custom match function? I think we can do everything above with just vanilla regex

Reply:
Alternatively, why not use https://docs.python.org/2.7/library/glob.html ?

Reply:
`glob`is solely for files; I looked into using [fnmatch](https://docs.python.org/2/library/fnmatch.html) to do the actual matching, but that's still file-centric. I wanted a simple matching function that matches on a single wildcard. I felt something small and specific was cleaner than a regex, but I can change it to a regex if you disagree.